projects/valtimo/dossier-management/src/lib/dossier-management-detail-container/tabs/dossier-management-search-fields/dossier-management-search-fields.component.ts

```diff
@@ -13,22 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import {Component} from '@angular/core';
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+  TemplateRef,
+  ViewChild,
+} from '@angular/core';
 import {DocumentService} from '@valtimo/document';
-import {BehaviorSubject, combineLatest, filter, map, Observable, switchMap, take, tap} from 'rxjs';
+import {
+  BehaviorSubject,
+  combineLatest,
+  filter,
+  map,
+  Observable,
+  Subscription,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs';
 import {ActivatedRoute} from '@angular/router';
-import {ListField} from '@valtimo/components';
+import {ListField, ModalComponent} from '@valtimo/components';
 import {TranslateService} from '@ngx-translate/core';
-import {DefinitionColumn, SearchField} from '@valtimo/config';
+import {
+  DefinitionColumn,
+  SearchField,
+  SearchFieldDataType,
+  SearchFieldFieldType,
+  SearchFieldMatchType,
+} from '@valtimo/config';
 import {DomSanitizer, SafeUrl} from '@angular/platform-browser';
+import {SelectItem} from '@valtimo/user-interface';
 
 @Component({
   selector: 'valtimo-dossier-management-search-fields',
   templateUrl: './dossier-management-search-fields.component.html',
   styleUrls: ['./dossier-management-search-fields.component.scss'],
 })
-export class DossierManagementSearchFieldsComponent {
+export class DossierManagementSearchFieldsComponent implements OnInit, OnDestroy, AfterViewInit {
+  @ViewChild('moveRowButtons') public moveRowButtonsTemplateRef: TemplateRef<any>;
+  @ViewChild('editSearchFieldModal') modal: ModalComponent;
```

Comment:
suggestion to change the variable name to a more semantic name, since this modal is not only for editing but is also used to create new search fields.
maybe rename `editSearchFieldModal` to just `searchFieldModal` or `searchFieldModalForm`